cmd/add.go

```diff
@@ -294,6 +348,91 @@ func fileDirIsInDirs(file string, dirSet map[string]bool) bool {
 	return false
 }
 
+// handleMeta takes the user provided meta and the backup meta inputs and
+// returns a map containing all valid metadata.
+func handleMeta(meta string, reason Reason, review, removal string) map[string]string {
+	userMeta := parseMetaString(meta)
+
+	mm, err := createBackupMetadata(reason, review, removal, userMeta)
+	if err != nil {
+		die(err.Error()) //nolint:govet
+	}
+
+	return mm
+}
+
+// createBackupMetadata returns a map containing the backup metadata values if
+// the provided inputs are valid.
+func createBackupMetadata(reason Reason, review, removal string, mm map[string]string) (map[string]string, error) {
+	review, removal = setReviewAndRemovalDurations(reason, review, removal)
+
+	removalDate, err := getFutureDateFromDuration(removal)
+	if err != nil {
+		return nil, err
+	}
+
+	reviewDate, err := getFutureDateFromDuration(review)
+	if err != nil {
+		return nil, err
+	}
+
+	if reviewDate.After(removalDate) {
+		die("--review duration must be smaller than --removal duration")
```

Comment:
Should probably be consistent on whether an error is returned or `die` is called; calling `die` is probably simpler, but returning an error is more go-like.